pulp_container/tests/functional/api/test_rbac_sync_repositories.py

```diff
@@ -1,98 +1,81 @@
-        with self.assertRaises(ApiException):
-            self.user3["repository_api"].partial_update(self.repository.pulp_href, body)
-        response = self.user1["repository_api"].partial_update(self.repository.pulp_href, body)
+    """Update a repository using HTTP PATCH."""
+    body = {"name": utils.uuid4()}
+    with user2, pytest.raises(ApiException):
+        container_repository_api.partial_update(repository.pulp_href, body)
+    with user3, pytest.raises(ApiException):
+        container_repository_api.partial_update(repository.pulp_href, body)
+    with user1:
+        response = container_repository_api.partial_update(repository.pulp_href, body)
         monitor_task(response.task)
-        type(self).repository = self.user1["repository_api"].read(self.repository.pulp_href)
+        repository = container_repository_api.read(repository.pulp_href)
 
-    @skip_if(bool, "repository", False)
-    def test_04_fully_update(self):
-        """Update a repository using HTTP PUT."""
-        body = {"name": utils.uuid4()}
-        with self.assertRaises(ApiException):
-            self.user2["repository_api"].update(self.repository.pulp_href, body)
-        with self.assertRaises(ApiException):
-            self.user3["repository_api"].update(self.repository.pulp_href, body)
-        response = self.user1["repository_api"].update(self.repository.pulp_href, body)
+    """Update a repository using HTTP PUT."""
+    body = {"name": utils.uuid4()}
+    with user2, pytest.raises(ApiException):
+        container_repository_api.update(repository.pulp_href, body)
+    with user3, pytest.raises(ApiException):
+        container_repository_api.update(repository.pulp_href, body)
+    with user1:
+        response = container_repository_api.update(repository.pulp_href, body)
         monitor_task(response.task)
-        type(self).repository = self.user1["repository_api"].read(self.repository.pulp_href)
+        repository = container_repository_api.read(repository.pulp_href)
 
-    @skip_if(bool, "repository", False)
-    def test_05_delete(self):
-        """Delete a repository."""
-        with self.assertRaises(ApiException):
-            self.user2["repository_api"].delete(self.repository.pulp_href)
-        with self.assertRaises(ApiException):
-            self.user3["repository_api"].delete(self.repository.pulp_href)
-        response = self.user1["repository_api"].delete(self.repository.pulp_href)
+    """Delete a repository."""
+    with user2, pytest.raises(ApiException):
+        container_repository_api.delete(repository.pulp_href)
+    with user3, pytest.raises(ApiException):
+        container_repository_api.delete(repository.pulp_href)
+    with user1:
+        response = container_repository_api.delete(repository.pulp_href)
         monitor_task(response.task)
-        with self.assertRaises(ApiException):
-            self.user1["repository_api"].read(self.repository.pulp_href)
+        with pytest.raises(ApiException):
```

Comment:
nice I didn't realize this was part of pytest.

Reply:
Well they sure needed to provide something like this.